pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -74,16 +69,11 @@ func (c *httpScrapeClient) Do(req *http.Request) (Stat, error) {
 	return statFromProto(resp.Body)
 }
 
-// isMeshError returns true if the error may have been caused by attempting to
-// access a pod directly when mesh is enabled.
-// TODO(julz) consider other tests, e.g. looking for upstream header in response.
-func isMeshError(err error) bool {
-	var errWithStatus errorWithStatusCode
-	if errors.As(err, &errWithStatus) {
-		return errWithStatus.statusCode == meshErrorStatusCode
-	}
-
-	return false
+// isPotentialMeshError returns whether the error encountered during scraping
+// is compatible with having been caused by the mesh being enabled.
+func isPotentialMeshError(err error) bool {
+	var se scrapeError
+	return errors.As(err, &se) && se.mightBeMesh
```

Comment:
added your bit of code golf from https://github.com/knative/serving/pull/11174#discussion_r612623785  here for you @vagababov  ;p